NEWS.md

```diff
@@ -2,12 +2,118 @@
+* New command `graphman drop` to hastly delete a subgraph deployment. [#4035](https://github.com/graphprotocol/graph-node/pull/4035)
+* New command `graphman chain call-cache` for clearing the call cache for a given chain. [#4066](https://github.com/graphprotocol/graph-node/pull/4066)
+* Add `--delete-duplicates` flag to `graphman check-blocks` by @tilacog in https://github.com/graphprotocol/graph-node/pull/3988
+
+#### Performance
+* Restarting a node now takes much less time because `postgres_fdw` user mappings are only rebuilt upon schema changes. If necessary, you can also use the new commands `graphman database migrate` and `graphman database remap` to respectively apply schema migrations or run remappings manually. [#4009](https://github.com/graphprotocol/graph-node/pull/4009), [#4076](https://github.com/graphprotocol/graph-node/pull/4076)
+* Database replicas now won't fall behind as much when copying subgraph data. [#3966](https://github.com/graphprotocol/graph-node/pull/3966) [#3986](https://github.com/graphprotocol/graph-node/pull/3986)
+* Block handlers optimization with Firehose >= 1.1.0. [#3971](https://github.com/graphprotocol/graph-node/pull/3971)
+* Reduced the amount of data that a non-primary shard has to mirror from the primary shard. [#4015](https://github.com/graphprotocol/graph-node/pull/4015)
+* We now use advisory locks to lock deployments' tables against concurrent writes. [#4010](https://github.com/graphprotocol/graph-node/pull/4010)
+
+#### Bug fixes
+* Fixed a bug which would cause some failed subgraphs to never restart. [#3959](https://github.com/graphprotocol/graph-node/pull/3959)
+* Fixed a determinism bug on subgraphs with `apiVersion < 0.0.5` using call handlers. [#4149](https://github.com/graphprotocol/graph-node/pull/4149)
+* Fixed a determinism bug that could arise when using `EXPERIMENTAL_STATIC_FILTERS`. [#4063](https://github.com/graphprotocol/graph-node/pull/4063)
+* Fixed a bug that would cause bad POIs for Firehose-backed subgraphs when processing `CREATE` calls. [#4085](https://github.com/graphprotocol/graph-node/pull/4085)
+* Fixed a bug which would cause failure to redeploy a subgraph immediately after deletion. [#4044](https://github.com/graphprotocol/graph-node/pull/4044)
+* Firehose connections are now load-balanced. [#4083](https://github.com/graphprotocol/graph-node/pull/4083)
+* Determinism fix. **See above.** [#4055](https://github.com/graphprotocol/graph-node/pull/4055)
+
+#### File data sources progress
+* File data sources now support directories in paths [#3992](https://github.com/graphprotocol/graph-node/pull/3992)
+* Exponential backoff, rate limiting for for file data sources. [#4117](https://github.com/graphprotocol/graph-node/pull/4117), [#4065](https://github.com/graphprotocol/graph-node/pull/4065)
+* Added support for the experimental `GRAPH_ALLOW_NON_DETERMINISTIC_IPFS` feature for file data sources. [#4011](https://github.com/graphprotocol/graph-node/pull/4011)
```

Comment:
Does this hide the functionality behind the same feature flag as the existing `ipfs.cat`?

Reply:
This actually has no effect on file data sources, it's just documenting the existing effect on `ipfs.cat`.

Reply:
Oversight on my part! Thanks for noticing, I removed the entire file data sources paragraph including this line. See https://github.com/graphprotocol/graph-node/pull/4199/commits/09c0ad48b1ece3df5e79a590dfd59a8f701a3ade.